walkeeper/src/safekeeper.rs

```diff
@@ -825,6 +829,45 @@ where
             self.append_response(),
         )))
     }
+
+    /// Update timeline state with peer safekeeper data.
+    pub fn record_safekeeper_info(&mut self, sk_info: &SafekeeperInfo) -> Result<()> {
+        let mut sync_control_file = false;
+        if let (Some(commit_lsn), Some(last_log_term)) = (sk_info.commit_lsn, sk_info.last_log_term)
+        {
+            // Note: the check is too restrictive, generally we can update local
+            // commit_lsn if our history matches (is part of) history of advanced
+            // commit_lsn provider.
+            if last_log_term == self.get_epoch() {
+                self.global_commit_lsn = max(commit_lsn, self.global_commit_lsn);
+                self.update_commit_lsn()?;
+            }
+        }
+        if let Some(s3_wal_lsn) = sk_info.s3_wal_lsn {
+            let new_s3_wal_lsn = max(s3_wal_lsn, self.inmem.s3_wal_lsn);
+            sync_control_file |=
+                self.inmem.s3_wal_lsn + (self.s.server.wal_seg_size as u64) < new_s3_wal_lsn;
+            self.inmem.s3_wal_lsn = new_s3_wal_lsn;
+        }
+        if let Some(remote_consistent_lsn) = sk_info.remote_consistent_lsn {
+            let new_remote_consistent_lsn =
+                max(remote_consistent_lsn, self.inmem.remote_consistent_lsn);
+            sync_control_file |= self.inmem.remote_consistent_lsn
+                + (self.s.server.wal_seg_size as u64)
+                < new_remote_consistent_lsn;
+            self.inmem.remote_consistent_lsn = new_remote_consistent_lsn;
+        }
+        if let Some(peer_horizon_lsn) = sk_info.peer_horizon_lsn {
+            let new_peer_horizon_lsn = max(peer_horizon_lsn, self.inmem.peer_horizon_lsn);
+            sync_control_file |= self.inmem.peer_horizon_lsn + (self.s.server.wal_seg_size as u64)
+                < new_peer_horizon_lsn;
+            self.inmem.peer_horizon_lsn = new_peer_horizon_lsn;
+        }
+        if sync_control_file {
+            self.persist_control_file()?;
```

Comment:
`persist_control_file` persists only `commit_lsn` and `peer_horizon_lsn`. Should we add `remote_consistent_lsn` and `s3_wal_lsn` there?